docs/instrumentation-list.yaml

```diff
@@ -322,6 +322,7 @@ couchbase:
 dropwizard:
   instrumentations:
   - name: dropwizard-metrics-4.0
+    disabledByDefault: true
```

Comment:
i'm realizing that im mixing camel and snake cases, and that we should probably stick to one convention. It looks like the collector uses [snake case](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/batchpersignal/metadata.yaml#L5), so perhaps it makes more sense to move to that

Reply:
since updating the names to use snake case will rewrite most of the instrumentation-list.yaml file, I'll do that change in a separate PR so it's easier to differentiate the other changes made in this PR